src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -1194,12 +1195,17 @@ export class ConnectionWidget extends lifecycle.Disposable {
 		let validInputs = this.validateInputs();
 		if (validInputs) {
 			if (this.useConnectionString) {
-				const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
-				if (connInfo) {
+				try {
+					const connInfo = await this._connectionManagementService.buildConnectionInfo(this.connectionString, this._providerName);
+					if (!connInfo) {
+						throw Error(localize('connectionWidget.ConnectionStringUndefined', 'No connection info returned.'));
+					}
 					model.options = connInfo.options;
 					model.savePassword = true;
-				} else {
-					this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"), localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string."));
+				} catch (err) {
+					this._logService.error(`${this._providerName} Failed to parse the connection string : ${err}`)
+					this._errorMessageService.showDialog(Severity.Error, localize('connectionWidget.Error', "Error"),
+						localize('connectionWidget.ConnectionStringError', "Failed to parse the connection string. {0}", utils.getErrorMessage(err)));
```

Comment:
I'd still suggest adding the stack trace (`err.stack`) as the message details - in the case that the error isn't an Error object/doesn't have a stack then it'll be undefined anyways so just result in no button which is fine. But the stack may contain useful information so would be good to add in still.